go/src/lib.rs

```diff
@@ -1,9 +1,166 @@
-/**
+/*
  * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
  */
-use std::ffi::c_void;
+#![deny(unsafe_op_in_unsafe_fn)]
 
+use glide_core::client::Client as GlideClient;
+use glide_core::connection_request;
+use glide_core::errors;
+use glide_core::errors::RequestErrorType;
+use glide_core::ConnectionRequest;
+use protobuf::Message;
+use std::{
+    ffi::{c_void, CString},
+    os::raw::c_char,
+};
+use tokio::runtime::Builder;
+use tokio::runtime::Runtime;
+
+/// Success callback that is called when a Redis command succeeds.
+///
+/// `index_ptr` is a baton-pass back to the caller language to uniquely identify the promise.
+/// `message` is the value returned by the Redis command.
+// TODO: Change message type when implementing command logic
+// TODO: Consider using a single response callback instead of success and failure callbacks
+pub type SuccessCallback = unsafe extern "C" fn(index_ptr: usize, message: *const c_char) -> ();
+
+/// Failure callback that is called when a Redis command fails.
+///
+/// `index_ptr` is a baton-pass back to the caller language to uniquely identify the promise.
+/// `error_message` is the error message returned by Redis for the failed command. It must be manually freed by the caller after this callback is invoked, otherwise a memory leak will occur.
```

Comment:
> It must be manually freed by the caller after this callback is invoked, otherwise a memory leak will occur.

if it's released by the caller, then the correct phrasing should be something like "The lifetime of `error_message` is until the callback returns the caller. The callee cannot use the data after the callback returns."